File: src/main/java/com/uml/generator/input/TraceReader.java
package com.uml.generator.input;

import com.uml.generator.model.Trace;
import java.util.List;

/**
 * Interface for reading Jaeger traces from various sources.
 */
public interface TraceReader {

    /**
     * Reads traces from the configured source.
     * 
     * @return list of traces
     * @throws Exception if reading fails
     */
    List<Trace> readTraces() throws Exception;
}

File: python_version/jaeger_uml_generator/input/trace_reader.py
"""Abstract base class for trace readers."""

from abc import ABC, abstractmethod
from typing import List
from ..models import Trace


class TraceReader(ABC):
    """Abstract base class for reading Jaeger traces from different sources."""
    
    @abstractmethod
    def read_traces(self) -> List[Trace]:
        """
        Read traces from the configured source.
        
        Returns:
            List of Trace objects
        """
        pass
